core/arch/arm/kernel/secure_partition.c

```diff
@@ -132,20 +125,19 @@ TEE_Result sp_partition_info_get(uint32_t ffa_vers, size_t buf_size,
 
 		if (s->state == sp_dead)
 			continue;
-		if (!count_only && count < count_limit) {
-			spmc_fill_partition_entry(ffa_vers, fpi,
-						  s->endpoint_id, 1,
-						  part_props, NULL);
-			fpi = (void *)((vaddr_t)fpi + fpi_size);
+		if (!count_only && !res) {
+			uint32_t uuid_words[4] = { 0 };
+
+			tee_uuid_to_octets((uint8_t *)uuid_words, ffa_uuid);
```

Comment:
`ffa_uuid` is NULL when we're querying all SPs, I think the argument should be `&s->ffa_uuid` instead.

Also it looks like these changes got mixed up during the squashing, shouldn't these be part of the previous commit instead?

Reply:
I'm sorry, I don't know what happened. I'll sort it out.

Reply:
> `ffa_uuid` is NULL when we're querying all SPs, I think the argument should be `&s->ffa_uuid` instead.

You're right, I'll fix it.

Reply:
Thanks, please squash the fixup.